### DISCUSSION TITLE
A strip chart from scatter plot : how to maintian a fast update rate

### DISCUSSION THREAD

**Discussion by avsteele** [LayTec-AG/Plotly.Blazor#205] [Q&A]
I'd like to use a Plotly to make a chart which can handle a few thousand points, and can be updated with new data at ~20 fps ideally. I am seeing some odd slowdowns and was wondering if someone can clue me into the cause and/or give suggestions for best practices.

I'm working from the example here:

https://github.com/LayTec-AG/Plotly.Blazor/blob/main/Plotly.Blazor.Examples/Components/ScatterChart.razor

To make a strip chart I added the code below. A button is attached to each of these functions. When one is pressed the chart's 1st series starts getting updated with a certain # of points at a certain period.  A rudimentary fps is also attached and displayed.

```cs
CancellationTokenSource? cts;
DateTime lastFrame = DateTime.Now;
private async Task ExtendContAsync(int count =15)
{
    try{
        cts = new();
        while(!(cts?.IsCancellationRequested ?? false))
        {
            // roughly 10 fame moving avg
            fps = fps*9/10 + 1 / (DateTime.Now - lastFrame).TotalSeconds/10;
            lastFrame = DateTime.Now;
            await ExtendData(count);
            await Task.Delay(30);
        }
    }
    finally
    {
        cts?.Dispose();
        cts = null;
    }
}

private void Stop()
{
    if (cts is not null) cts.Cancel();
}
```

This can run at ~15 fps and accumulates ~6K points without a problem.  Removing amrkers, and only having lines results in a small speedup, unsurprisingly. 

The odd behavior is: If I let this run, then stop it, "Pop'' the series, "Push" a new one (returning the chart visually to its initial state) and start it again I see a slowdown to below 10 fps after a few hundred points are displays.  This slowdown is never present on the first run, and **only presents when the series uses markers**.  I can't see what cold cause this in the razor code itself.

Thanks for any thoughts.

**Comment by sean-mcl** [comment]
Hey, your code looks good so far.

Have you tried your performance test using the javascript version of plotly? Would be interesting if the wrapper or plotly.js is the bottleneck.

**Reply by avsteele** [reply]
Good idea but I don't have enough JS experience to test it.

I can confirm this slowdown behavior is observed in both Firefox and Brave.
